src/main/java/teammates/common/util/EmailLogEntry.java

```diff
@@ -0,0 +1,147 @@
+import java.util.TimeZone;
+import javax.mail.Message;
+import javax.mail.internet.MimeMessage;
+
+import com.google.appengine.api.log.AppLogLine;
+
+/** A log entry which contains info about subject, receiver, content and sent date of a sent email*/
+public class EmailLogEntry {
+    
+    private String receiver;
+    private String subject;
+    private String content;
+    private long time;
+    
+    public String logInfoAsHtml;
+    
+    public EmailLogEntry(MimeMessage msg) throws Exception{
+        
+            this.receiver = msg.getRecipients(Message.RecipientType.TO)[0].toString();
+            this.subject = msg.getSubject();
+            this.content = (String) msg.getContent(); 
+    }
+    
+    public EmailLogEntry(AppLogLine appLog){
+        this.time = appLog.getTimeUsec() / 1000;
+        String[] tokens = appLog.getLogMessage().split("\\|\\|\\|", -1);
+        
+        try{
+            this.receiver = tokens[1];
+            this.subject = tokens[2];
+            this.content = tokens[3];
+            logInfoAsHtml = getLogInfoForTableRowAsHtml();
+        } catch (ArrayIndexOutOfBoundsException e){
+            this.receiver = "";
+            this.subject = "";
+            this.content = "";
+            logInfoAsHtml = "";
+        }
+    }
+    
+    private String getLogInfoForTableRowAsHtml(){
+
+        StringBuilder str = new StringBuilder();
+        str.append("<tr class=\"log\">");
+        str.append("<td>" + this.receiver + "</td>");
+        str.append("<td>" + this.subject + "</td>");
+        str.append("<td>" + this.getTimeForDisplay() + "</td>");
+        str.append("</tr>");
+        str.append("<tr id=\"small\">");
+        str.append("<td colspan=\"3\">");
+        str.append("<ul class=\"list-group\">");
+        str.append("<li class=\"list-group-item list-group-item-info\">");
+        str.append("<input type=\"text\" value=\"" + this.getContent() + "\" class=\"form-control\"");
+        str.append(" readonly=\"readonly\">");
+        str.append("</li>");
+        str.append("</ul>    ");
+        str.append("</td>");
+        str.append("</tr>");
+        str.append("<tr id=\"big\" style=\"display:none;\">");
+        str.append("<td colspan=\"3\">");
+        str.append("<div class=\"well\">");
+        str.append("<ul class=\"list-group\">");
+        str.append("<li class=\"list-group-item list-group-item-success\"><small>");
+        str.append(this.content + "</small>");
+        str.append("</li>");
+        str.append("</ul>");
+        str.append("</div>");
+        str.append("</td>");
+        str.append("</tr>");
+        
+        return str.toString();
+    }
+    
+    /**
+     * Generates a log message that will be logged in the server
+     */
+    public String generateLogMessage(){
+        //TEAMMATESEMAILSLOG|||RECEIVER|||SUBJECT|||CONTENT|||DATE
+        return "TEAMMATESEMAILLOG|||" + this.receiver + "|||" + 
+               this.subject + "|||" + this.content;
```

Comment:
You  might have to truncate the content. I think a log entry has a length limit.